packages/core/src/routes/interaction/index.ts

```diff
@@ -48,6 +51,42 @@ export default function interactionRoutes<T extends AnonymousRouter>(
     }
   );
 
+  router.patch(
+    identifierPrefix,
+    koaInteractionBodyGuard(),
+    koaSessionSignInExperienceGuard(provider),
+    async (ctx, next) => {
+      const { event } = ctx.interactionPayload;
+      const interactionStorage = await getInteractionStorage(ctx, provider);
+
+      // Forgot Password specific event interaction can't be shared with other types of interactions
+      assertThat(
+        event === Event.ForgotPassword
+          ? interactionStorage.event === Event.ForgotPassword
+          : interactionStorage.event !== Event.ForgotPassword,
+        new RequestError({ code: 'session.verification_session_not_found' })
+      );
```

Comment:
This looks odd... Can other event types share the interactions between each other? For example, `event` is SignIn, and `interactionStorage.event` is Register.. Would this be a valid case?

Reply:
yes, sign-in and register can be shared and updated by patch event.